gitlab-mr-status/src/main/java/com/tw/go/plugin/provider/GitLabProvider.java

```diff
@@ -82,14 +87,15 @@ public void updateStatus(String url, PluginSettings pluginSettings, String prIdS
 
         GitlabAPI api = GitlabAPI.connect(endPointToUse, oauthAccessTokenToUse);
         GitlabProject project = api.getProject(getRepository(url));
-        api.createCommitStatus(project, revision, GitLabState.stateFor(result), prIdStr, "GoCD", trackbackURL, "");
+        String state = GitLabState.stateFor(result);
+        api.createCommitStatus(project, revision, state, prIdStr, "GoCD", trackbackURL, "");
     }
 
     private String getValueFromPluginSettings(Object values) {
         if (values == null) {
             return "";
         }
-        Map<String, String> vals = (LinkedHashTreeMap<String, String>)values;
+        Map<String, String> vals = (LinkedTreeMap<String, String>) values;
```

Comment:
I had to change this because, on GoCD 19.12 this was throwing an error as InvalidClassCast Exception.